src/member/get-member-query/member-query-observer.ts

```diff
@@ -0,0 +1,147 @@
+import { EntityManager, FindOptionsWhere, SelectQueryBuilder } from "typeorm";
+import { Member } from "../entity/member.entity";
+import { MemberProperty } from "../entity/member_property.entity";
+import { first, keys, pick, values } from "lodash";
+import { MemberPropertiesCondition } from "~/member/member.dto";
+import { MemberPermissionGroup } from "../entity/member_permission_group.entity";
+import { MemberCategory } from "../entity/member_category.entity";
+import { MemberTag } from "../entity/member_tag.entity";
+import { MemberPhone } from "../entity/member_phone.entity";
+
+class MemberManagerObserver implements QueryObserver {
+  update(entityManager: EntityManager, queryBuilder: SelectQueryBuilder<Member>, conditions: FindOptionsWhere<Member>): void {
```

Comment:
Promise<void> ?

Reply:
@Ponchimeow 

The return type of the update method depends on whether it contains asynchronous operations. If it does, it should be marked as async and return a Promise<void>; otherwise, it can return void

MemberManagerObserver is responsible for filtering the manager field in the original member table, and since it doesn't involve any asynchronous operations, it doesn't need to return a Promise